diffsims/pattern/detector_functions.py

```diff
@@ -0,0 +1,240 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2020 The diffsims developers
+#
+# This file is part of diffsims.
+#
+# diffsims is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# diffsims is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with diffsims.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+
+from numpy.random import default_rng
+from scipy import ndimage as ndi
+
+
+def _process_seed_argument(seed):
+    """ Sets up a numpy random number generator with a seed"""
+    if seed is not None:
+        rng = default_rng(seed)
+    else:
+        rng = default_rng()
+
+    return rng
+
+
+def constrain_to_dynamic_range(pattern, detector_max=None):
+    """Force the values within pattern to lie between [0,detector_max]
+
+    Parameters
+    ----------
+    pattern : np.array
+        The diffraction pattern at the detector after corruption
+    detector_max : float
+        The maximum allowed value at the detector
+
+    Returns
+    -------
+    within_range_pattern: np.array
+        The pattern, with values >=0 and =< detector_max
+    """
+    pattern[pattern < 0] = 0
+
+    if detector_max is not None:
+        pattern[pattern > detector_max] = detector_max
+
+    return pattern
+
+
+def add_gaussian_blur(pattern, sigma):
```

Comment:
for consistency with other parts of this PR now - should this be `add_gaussian_point_spread`?